pilot/pkg/model/authentication.go

```diff
@@ -230,20 +230,14 @@ func GetAmbientPolicyConfigName(key ConfigKey) string {
 func getConfigsForWorkload(rootNamespace string, configsByNamespace map[string][]config.Config, selectionOpts WorkloadPolicyMatcher) []*config.Config {
 	workloadLabels := selectionOpts.WorkloadLabels
 	namespace := selectionOpts.WorkloadNamespace
-	serviceNamespace := selectionOpts.ServiceNamespace
 	configs := make([]*config.Config, 0)
-	var lookupInNamespaces []string
-	if namespace != rootNamespace {
-		// Only check the root namespace if the (workload) namespace is not already the root namespace
-		// to avoid double inclusion.
-		lookupInNamespaces = []string{namespace, rootNamespace}
-	} else {
-		lookupInNamespaces = []string{namespace}
+	lookupInNamespaces := sets.New[string]()
+	lookupInNamespaces.Insert(namespace)
+	lookupInNamespaces.Insert(rootNamespace)
+	for _, svc := range selectionOpts.Services {
+		lookupInNamespaces.Insert(svc.Namespace)
 	}
-	if serviceNamespace != "" && !slices.Contains(lookupInNamespaces, serviceNamespace) {
-		lookupInNamespaces = append(lookupInNamespaces, serviceNamespace)
-	}
-	for _, ns := range lookupInNamespaces {
+	for _, ns := range lookupInNamespaces.UnsortedList() {
```

Comment:
Hmm.. is this non-determinism an issue?

The resulting config list is non-deterministic ordering

GetJwtPoliciesForWorkload: not a problem, it either inserts into a Set or does SortBy(GetIssuer)

GetPeerAuthenticationsForWorkload: Sorts by creation timestamp (also, PA will never have .Services set)

So all current cases it shouldn't be, but it is a possible risk for sure.